pyproject.toml

```diff
@@ -11,6 +11,9 @@ exclude = '''
 )
 '''
 
+[tool.pytest.ini_options]
+addopts = "--import-mode=append"
+
```

Comment:
I have never seen this. how come it's not needed for Hydra?

Reply:
That surprised me as well. Just looked into it more closely and this is because of this line in setup.py:
```
entry_points={"pytest11": ["hydra_pytest = hydra.extra.pytest_plugin"]},
```

Because of this, the `hydra` module is imported from its pip install location *before* `pytest` prepends the local folder to `sys.path` (this is `pytest`'s default behavior, see https://docs.pytest.org/en/stable/pythonpath.html). So the tests run against the installed version as intended, but essentially "by luck".

Reply:
oh, nasty. I was wondering if the difference in behavior had anything to do with it.
That pytest plugin is fairly recent (maybe a couple of months).

How did you initially notice it was running against the local version and not the installed version? shouldn't the local version also have the generated parser code?

Reply:
> How did you initially notice it was running against the local version and not the installed version? shouldn't the local version also have the generated parser code?

No when you run `pip install .` the parser code is not generated locally, ony in the installed version.

Reply:
When running `pip install .`, the parser is generated while making the sdist wheel which in turn is installed.
why do you need the generated code in this scenario? it's installed and available.

Reply:
> When running `pip install .`, the parser is generated while making the sdist wheel which in turn is installed.
> why do you need the generated code in this scenario? it's installed and available.

So, starting from a clean repo, running `pip install .` generates the grammar code and installs it somewhere in a path that looks like `/Users/odelalleau/opt/miniconda3/envs/py38-omega/lib/python3.8/site-packages/omegaconf`. However, if you look at the content of `./omegaconf/grammar/gen`, this folder is still emtpy.

Without this change to the pytest config, pytest would use `./omegaconf` instead of `/Users/odelalleau/opt/miniconda3/envs/py38-omega/lib/python3.8/site-packages/omegaconf` and thus wouldn't be able to find the generated files.

Reply:
> So, starting from a clean repo, running pip install . generates the grammar code and installs it somewhere in a path that looks like /Users/odelalleau/opt/miniconda3/envs/py38-omega/lib/python3.8/site-packages/omegaconf. However, if you look at the content of ./omegaconf/grammar/gen, this folder is still emtpy.

This is what I am expecting.
But for starters, the current logic in nox (master) is just running pytest in the dir.
In Hydra it also `pip install .` (or `pip install -e .`).
Can you try that and see if eliminates the need for this hack?
(again, trying to control the creeping complexity here).

Reply:
Let's continue this thread in https://github.com/omry/omegaconf/pull/321#discussion_r513670942

Reply:
Actually it's not. this is a different issue.

Reply:
> Actually it's not. this is a different issue.

I'll follow-up about this comment in the new upcoming PR.